### ISSUE TITLE
Feature Request - after script for vagrant up

### ISSUE THREAD

**Issue by skylerkatz** [laravel/homestead#1401]
Hello!

For a recent project I needed to run some shell scripts each time I started the vagrant machine.

This ended up being pretty easy to implement, but there was not a lot documented around the web about how to do this.

Would you be open to a PR that adds this functionality?

Obviously, you don't have to accept it if I do submit the pr, but wanted to check in before extracting the pr.

Thanks!

**Comment by svpernova09** [comment]
Hi! :D

https://laravel.com/docs/7.x/homestead#extending-homestead

We have `after.sh` (Should be in VCS) and `user-customizations.sh` (Shouldn't be in VCS) to solve this. If what you build would add on these I'd be interested.

I'm always curious about how others implement Homestead.

**Comment by skylerkatz** [comment]
I make pretty heavy usage of after.sh to install other os level packages, but there were a handful of things that I need to run every time I run `vagrant up`.

In my case, I need to make sure that Apache is running and that some os packages start.

I will go ahead and make a quick pr that adds a little more context.

**Comment by svpernova09** [comment]
AH! I missed the `each time I started the vagrant machine`.

> make sure that Apache is running

If you're using site type `apache` it should be running, if not that's a bug and we should ensure that script enables the service.

> that some os packages start

This is something you should be using `service` for. Maybe even `supervisorctl`. 


I'll have a look at your PR later (Short on time atm).

**Comment by skylerkatz** [comment]
No rush on the PR.

There may be a bug, but it likely has to do with SSL certificate generation and not with the site type.

In my use case I am running a handful of api services that all communicate with eachother. One of the services requires wildcard subdomains for per account access. For some reason, the only way to generate a proper wildcard SSL cert, I needed to set the site up twice in homestead.yaml. Once as apache and then again as laravel.
```yaml
-
    map: "my-app.test"
    to: /home/vagrant/repositories/my-app/public
    type: "apache"
    php: "7.3"

    # Used to auto generate the proper wildcard SSL Cert for Sermon Cloud
-
    map: "*.my-appo.test"
    to: /home/vagrant/repositories/my-app
    type: "laravel"
    php: "7.3"
```
When vagrant starts, it sees the laravel site and turns on ngnix. I unfortunately cannot use nginx (even though most of the projects on the box are laravel) because our main app is a legacy project that requires apache.

So I generate both sites and then in the `after.sh` script I make some changes to the apache config to point to the wildcard cert.

In terms of the OS packages, I did look into `service` and `supervisorctl` but I did not find that they fit my use case because I need to pipe in results from database queries and this felt like the easiest route.

Anyway, thanks again!

**Comment by svpernova09** [comment]
I think there are a few issues here. First I want to support wildcard certs for real without doubling site creation. #1410 is the first step towards this (nginx working atm, apache next).

I *think* this may solve part of the issue.

The next part of the issue I believe is that we're not properly disabling the Nginx service and enabling the apache service. I'll dig into this and see if I can find the cause of apache not starting properly.

**Comment by svpernova09** [comment]
I just merged #1410 into master. I haven't tagged a release yet. It would be awesome if you could give this version a spin. It should have solved your issues with "make sure that Apache is running" and also creating wildcard certificates. 

The only thing left to address would be the OS packages you need to start on boot, which my first recommendation would be to use cron's `@reboot` syntax to start them

**Comment by skylerkatz** [comment]
Hi @svpernova09,

Sorry it took so long to get back to you. Time seems to get away from me these days.

Your SSL updates have solved the Apache issues, so thank you so much for that.

I will take a look at the `@reboot` syntax and see if it accomplishes what I am looking for.

Thanks again for all of your hard work on this project. I use it daily and have learned a lot about how servers work by looking through the repository.
